guides/plugins/plugins/administration/add-rule-assignment-configuration.md

```diff
@@ -0,0 +1,256 @@
+
+{% code title="Example of a one-to-many deleteContext" %}
+```
+deleteContext: {
+    type: 'one-to-many',
+    entity: 'cms_block',
+    column: 'extensions.swagCmsExtensionsBlockRule.visibilityRuleId',
+},
+```
+{% endcode %}
+
+The other type is `many-to-many`, which has to link to the `ManyToManyAssociationField` of the extension like this:
+
+{% code title="Example of a many-to-many deleteContext" %}
+```
+deleteContext: {
+    type: 'many-to-many',
+    entity: 'category',
+    column: 'extensions.swagDynamicAccessRules',
+},
+```
+{% endcode %}
+
+### Provide to add an assignment
+
+If you want to provide to add an assignment, you have to define the `addContext`. This context has the same two types as the `deleteContext` (see above),
+but the `addContext` has more option to fill out, because an add assignment modal has to be configured:
+
+{% code title="Example of a one-to-many addContext" %}
+```
+addContext: {
+    type: 'one-to-many',
+    entity: 'shipping_method',
+    column: 'availabilityRuleId',
+    searchColumn: 'name',
+    criteria: () => {
+        const criteria = new Criteria();
+        criteria.addFilter(Criteria.not(
+            'AND',
+            [Criteria.equals('availabilityRuleId', ruleId)],
+        ));
+
+        return criteria;
+    },
+    gridColumns: [
+        {
+            property: 'name',
+            label: 'Name',
+            rawData: true,
+            sortable: true,
+            allowEdit: false,
+        },
+        {
+            property: 'description',
+            label: 'Description',
+            rawData: true,
+            sortable: true,
+            allowEdit: false,
+        },
+        {
+            property: 'taxType',
+            label: 'Tax calculation',
+            rawData: true,
+            sortable: true,
+            allowEdit: false,
+        },
+        {
+            property: 'active',
+            label: 'Active',
+            rawData: true,
+            sortable: true,
+            allowEdit: false,
+        },
+    ],
+},
+```
+{% endcode %}
+
+The `addContext` needs a definition of the `gridColumns`, the `entity` and the `criteria`, like in the general configuration.
+Also, the context needs the `column` of the assignment and the `searchColumn` of the assigned entity.
+
+A context of the `many-to-many` type would look like this:
+
+{% code title="Example of a many-to-many addContext" %}
+```
+addContext: {
+    type: 'many-to-many',
+    entity: 'swag_dynamic_access_category_rule',
+    column: 'categoryId',
+    searchColumn: 'name',
+    association: 'swagDynamicAccessRules',
+    criteria: () => {
+        const criteria = new Criteria();
+        criteria.addFilter(Criteria.equals('parentId', null));
+
+        return criteria;
+    },
+    gridColumns: [
+        // Definition of columns
+    ],
+},
+```
+{% endcode %}
+
+Beside the properties of a `one-to-many` type you have to define the `association` with the name of the `ManyToManyAssociationField`.
```

Comment:
You could add a "## Further reading" heading in the bottom and add a reference to - https://developer.shopware.com/docs/guides/plugins/plugins/framework/rule/add-custom-rules

If you have any other interesting cross-reference, feel free to put them here (or link from existing articles to the one that was created).